cfme/containers/node.py

```diff
@@ -2,21 +2,69 @@
 # added new list_tbl definition
 from functools import partial
 from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic.widget import View, Text
+from widgetastic_manageiq import (
+    BootstrapSelect, Button, Table, Accordion, ManageIQTree, PaginationPane)
+from widgetastic_patternfly import Dropdown
 
+from cfme import BaseLoggedInPage
 from cfme.common import SummaryMixin, Taggable
+from cfme.containers.provider import ContainersProvider
+from cfme.exceptions import NodeNotFound
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import CheckboxTable, toolbar as tb, paginator, InfoBlock, match_location
+from cfme.web_ui import CheckboxTable, toolbar as tb, InfoBlock, match_location
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from utils.log import logger
 from . import pol_btn, mon_btn
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
 
 match_page = partial(match_location, controller='container_node', title='Nodes')
 
+resource_locator = "//div[@id='records_div']/table//span[@title='{}']"
 
-class Node(Taggable, SummaryMixin, Navigatable):
 
+class NodeView(BaseLoggedInPage):
+    title = Text('#explorer_title_text')
+
+    monitor = Dropdown('Monitoring')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+
+    list = Table(locator="//div[@id='list_grid']//table")
+
+    @property
+    def in_cloud_instance(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Compute', 'Containers', 'Container Nodes'] and
+            match_page()  # No summary, just match controller and title
+        )
+
+
+class NodeCollection(Navigatable):
+    """Collection object for :py:class:`Node`."""
+
+    def instantiate(self, name, provider):
+        return Node(name=name, provider=provider, appliance=self.appliance)
+
+    def all(self):
+        # container_nodes table has ems_id, join with ext_mgmgt_systems on id for provider name
+        node_table = self.appliance.db['container_nodes']
+        ems_table = self.appliance.db['ext_management_systems']
+        node_query = self.appliance.db.session.query(node_table.name, ems_table.name)\
+            .join(ems_table, node_table.ems_id == ems_table.id)
+        nodes = []
+        for name, provider_name in node_query.all():
+            # Hopefully we can get by with just provider name?
+            nodes.append(Node(name=name,
+                              provider=ContainersProvider(name=provider_name,
+                                                          appliance=self.appliance)))
```

Comment:
Looking at this made me think - shall we ensure that provider with name A will be the same object for all the things returned by this method.

Thoughts? @psav @mshriver @RonnyPfannschmidt